packages/framework/src/evaluate/evaluate.ts

```diff
@@ -38,14 +50,27 @@ export const buildEvaluateFn = (
       // Need to filter out invalid JS identifiers
     ].filter(([key, _]) => !key.includes(".")),
   );
-  const globalBlock = screen.model?.global;
-  const importedScriptBlock = screen.model?.importedScripts;
+
+  if (has(invokableObj, "ensemble")) {
+    const tempEnsemble = get(invokableObj, "ensemble") as {
+      [key: string]: unknown;
+    };
+    (window as unknown as InvokableWindow).ensemble = omitBy(
+      tempEnsemble,
+      isUndefined,
+    );
+  }
```

Comment:
This can be set once - most of the functions in ensemble API do not require any screen params. The only thing would be the modal context, which can be passed in as an argument.